### DISCUSSION TITLE
How can I add my own superatoms into AbbreviationsDefinition

### DISCUSSION THREAD

**Discussion by mily33** [rdkit/rdkit#3677] [Q&A] [answered]
The function rdkit.Chem.rdAbbreviations.GetDefaultAbbreviations() return a list of 37 predefined superatoms. I am trying to add those superatoms defined in [superatom.txt](https://github.com/metamolecular/osra/blob/master/dict/superatom.txt) into the default abbreviation definition. How can I do this? 

Besides, when I print the *.mol for the default abbreviation, I got this error:

Traceback (most recent call last):
  File "\<stdin\>", line 1, in <module>
TypeError: No to_python (by-value) converter found for C++ type: std::shared_ptr<RDKit::ROMol>

code is:
`defaults = rdkit.Chem.rdAbbreviations.GetDefaultAbbreviations()`
`a = defaults[0]`
`a.mol`

**Comment by greglandrum** [comment, marked as answer]
Hi @mily33,

Here's a quick example demonstrating how to reformat that file and use it to create a set of abbreviations:
https://gist.github.com/greglandrum/c2df5e0ad91f4eee050848554264eff7

Note that this does not handle the last entries. Since those are defined by a dummy atom (which would match anything) they aren't suitable for use as abbreviations (though they probably could be used to expand superatoms in input data).

The problem with `a.mol` giving an error is a bug. Would you mind reporting that as a separate issue?

-greg

**Reply by mily33** [reply]
Thanks! It solved my problem.
And I will report the problem with a.mol as a separate issue.

**Reply by greglandrum** [reply]
awesome. Glad to hear it!
